### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by MohammedBuirat**
Implemented using strategy pattern for handling the different types of data 
Observer to notify all the bots when a reading occur

### CODE REVIEW

File: weeather/LoadBots.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    internal class LoadBots
+    {
+
+        public List<Bot> LoadBotsFromConfigFile()
+        {
+            string data = ConfigurationFileData();
+            return ParseToBots(data);
+        }
+
+        private string ConfigurationFileData()
```
Comment (WadeeSamiAZ) [inline weeather/LoadBots.cs]: let's rename the method `loadConfigFile`

File: weeather/LoadBots.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            string fileData = File.ReadAllText("../../../config.json");
+            return fileData;
+        }
+
+        //here i had to do it like this since i will not be able to pass the bot in 
+        //bitConfiguration to a new method which caused me to preform all the work in a single method
+        private List<Bot> ParseToBots(string data)
+        {
+            List<Bot> bots = new List<Bot>();
+            var botConfigurations = JsonSerializer.Deserialize<Dictionary<string, JsonElement>>(data);
```
Comment (WadeeSamiAZ) [inline weeather/LoadBots.cs:26]: I can see that the logic here handles json only, the requirements of the task states that configs can either be json or xml, can you please handle this ?

File: weeather/LoadBots.cs (hunk 1 above)
Comment (MohammedBuirat) [reply weeather/LoadBots.cs]: Can't agree more

File: weeather/LoadBots.cs (hunk 2 above)
Comment (MohammedBuirat) [reply weeather/LoadBots.cs:26]: Aren't the different file formats for weather data? It says that the configuration file will be in JSON format 
quoting  "The configuration file should be in a JSON format"

**Pull request merged**